libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -368,6 +364,7 @@ class MapboxNavigation(
         navigationSession.unregisterAllNavigationSessionStateObservers()
         fasterRouteController.stop()
         routeRefreshController.stop()
+        MapboxNavigationTelemetry.unregisterListeners(this@MapboxNavigation)
```

Comment:
Could you add a unit test?

Reply:
Indeed. We were discussing that internally 😅 

> how can we make sure we don't mess it up in the future?
> maybe we can add a test verifying the order

Reply:
Added a couple 👍 

https://github.com/mapbox/mapbox-navigation-android/blob/b8ee056487e06a370f66afd0f8a65368ad688f1d/libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt#L358-L381